rcl/src/rcl/node.c

```diff
@@ -383,6 +386,15 @@ rcl_node_init(
       }
       allocator->deallocate(node->impl->graph_guard_condition, allocator->state);
     }
+    if (NULL != node->impl->options.arguments.impl) {
```

Comment:
nitpick: You could adhere to the same style of the file/function
```
if (node->impl->options.arguments.impl) {
...
}
```

Reply:
As we're trying to move closer to MISRA, I encourage the opposite (keeping the current state of this PR) as all conditions need to evaluate explicitly as boolean

Reply:
i understand. Yet I believe we should either have a misra check on the complete file or not.
but as I said, just a nitpick. No show-stopper here from my side.